src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -35,16 +30,25 @@
     </TestTargetFramework>
   </ItemGroup>
 
-  <Target Name="CopyTestToTestDirectory"
-    Inputs="@(PackagesConfig);$(TestRuntimePackageConfig);@(ReferenceCopyLocalPaths);@(Content);@(IntermediateAssembly);@(_DebugSymbolsIntermediatePath)"
-    Outputs="$(TestPath)%(TestTargetFramework.Folder)\*.*"
-    Condition="'$(CopyTestToTestDirectory)'=='true'">
-    
+  <ItemGroup>
+    <PackagesConfigs Include="$(ProjectPackagesConfigFile)" />
+    <PackagesConfigs Include="$(TestRuntimePackageConfig)" />
+  </ItemGroup>
+
+  <Target Name="DiscoverTestInputs" BeforeTargets="RunTestsForProject">
     <ItemGroup>
-      <PackagesConfigs Include="$(ProjectPackagesConfigFile)" />
-      <PackagesConfigs Include="$(TestRuntimePackageConfig)" />
+      <TestInputs Include="@(PackagesConfigs)" />
```

Comment:
1) If you are asking if the DiscoverTestInputs target should be moved to tests.targets...given the even stronger coupling between RunTestsForProject and CopyTestToTestDirectory, I am not comfortable with the separation between tests.targets and publishtest.targets. It made sense when they were happening at two different points in the lifecycle, but now it just feels strange. I have been dancing around combining them.
2) Agreed, I will rename it.